tesseract-server/src/handlers/flush.rs

```diff
@@ -1,7 +1,65 @@
 use serde_derive::{Serialize, Deserialize};
 
+use lazy_static::lazy_static;
+use log::*;
+use serde_qs as qs;
+use std::env;
+
+use actix_web::{
+    HttpRequest,
+    HttpResponse,
+    Result as ActixResult,
+};
+
+use crate::app::AppState;
+use crate::schema_config;
+
 #[derive(Debug, Deserialize, Serialize)]
 pub struct FlushQueryOpt {
     pub secret: String,
 }
 
+pub fn flush_handler(req: HttpRequest<AppState>) -> ActixResult<HttpResponse> {
+    let query = req.query_string();
+    lazy_static!{
+        static ref QS_NON_STRICT: qs::Config = qs::Config::new(5, false);
+    }
+    let query_res = QS_NON_STRICT.deserialize_str::<FlushQueryOpt>(&query);
+    let query = match query_res {
+        Ok(q) => q,
+        Err(err) => {
+            return Ok(HttpResponse::BadRequest().json(err.to_string()));
+        },
+    };
+
+    let db_secret = match env::var("TESSERACT_FLUSH_SECRET") {
```

Comment:
Let's move this into the `main` module, with the other config stuff. That will also prevent `env::var` from being called on every request.

Reply:
This might require holding the secret in `AppState`. Perhaps using `Option`, in case we don’t want to allow not flushing when there’s no secret?